src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -547,41 +547,116 @@ private void SetDirty(TimeSpan? maxDelay)
             return applicationData;
         }
 
-        private async Task InitializeAsync(bool ignoreFailures, IEnumerable<ConfigurationClient> availableClients, CancellationToken cancellationToken = default)
+        public async Task LoadAsync(bool ignoreFailures, CancellationToken cancellationToken)
         {
-            Dictionary<string, ConfigurationSetting> data = null;
-            Dictionary<KeyValueIdentifier, ConfigurationSetting> watchedSettings = null;
-            
+            var startupExceptions = new List<Exception>();
+
+            int attempts = 0;
+
             try
             {
-                await ExecuteWithFailOverPolicyAsync(
-                    availableClients,
-                    async (client) =>
+                while (true)
+                {
+                    attempts++;
+
+                    IEnumerable<ConfigurationClient> clients = _configClientManager.GetAvailableClients(DateTimeOffset.UtcNow, ignoreBackoff: true);
+
+                    if (await TryInitializeAsync(clients, startupExceptions, cancellationToken).ConfigureAwait(false))
                     {
-                        data = await LoadSelectedKeyValues(
-                            client,
-                            cancellationToken)
-                            .ConfigureAwait(false);
-
-                        watchedSettings = await LoadKeyValuesRegisteredForRefresh(
-                            client,
-                            data,
-                            cancellationToken)
-                            .ConfigureAwait(false);
-
-                        watchedSettings = UpdateWatchedKeyValueCollections(watchedSettings, data);
-                    },
-                    cancellationToken)
-                    .ConfigureAwait(false);
+                        break;
+                    }
+
+                    try
+                    {
+                        await Task.Delay(FailOverConstants.MinStartupBackoffDuration.CalculateBackoffDuration(FailOverConstants.MaxBackoffDuration, attempts), cancellationToken).ConfigureAwait(false);
+                    }
+                    catch (OperationCanceledException)
+                    {
+                        throw new TimeoutException($"The provider timed out while attempting to load data on startup.",
```

Comment:
```suggestion
                        throw new TimeoutException($"The provider timed out while attempting to load.",
```

I would leave out "on startup", technically nothing prevents someone from instantiating our provider during the lifetime of the application.